synapse/push/bulk_push_rule_evaluator.py

```diff
@@ -491,7 +491,6 @@ async def _action_for_event_by_user(
 
 def _flatten_dict(
     d: Union[EventBase, Mapping[str, Any]],
-    room_version: Optional[RoomVersion] = None,
```

Comment:
The event has the room version in it; I think this change will mean that the related events stuff (which is also experimental) will start running the extensible events behavior too. Both of these are experimental and it is undefined how they should interact, but that seems OK to me.